mempool/src/shared_mempool/tasks.rs

```diff
@@ -350,6 +351,87 @@ where
     statuses
 }
 
+/// Submits a list of SignedTransaction to the local mempool
+/// and returns a vector containing SubmissionStatusBundle.
+#[cfg(feature = "consensus-only-perf-test")]
+pub(crate) fn process_incoming_transactions<V>(
```

Comment:
nit: can we reduce the duplication between these two? e.g., define an internal helper that does most of the identical logic?